src/protocol/vm/state_builder.rs

```diff
@@ -386,9 +389,7 @@ impl VMPoolStateBuilder {
         let method_name = decoded
             .split(':')
             .last()
-            .ok_or_else(|| {
-                SimulationError::DecodingError("Invalid decoded string format".into())
-            })?;
+            .ok_or_else(|| SimulationError::FatalError("Invalid decoded string format".into()))?;
```

Comment:
Can we add more detail here too? We're expecting it to be formatted as an address no?